File: sanghyun/programmers-2/하노이의 탑.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    ArrayList<int[]> list;
    public int[][] solution(int n) {
        list = new ArrayList<>();
        hanoi(n, 1, 2, 3);
        int[][] answer = new int[list.size()][2];

        for (int i = 0; i < list.size(); i++) {
            answer[i][0] = list.get(i)[0];
            answer[i][1] = list.get(i)[1];
        }
        return answer;
    }
    public void hanoi(int n, int start, int mid, int to) {
        if (n == 1) {
            list.add(new int[]{start, to});
            return;
        }

        hanoi(n - 1, start, to, mid);
        list.add(new int[]{start, to});
        hanoi(n - 1, mid, start, to);
    }
}

File: sanghoon/programmers-2/하노이의 탑.py
def solution(n):
    def hanoi(n, first, last, mid):
        if n == 1:
            answer.append([first, last])
            return
        hanoi(n - 1, first, mid, last)
        answer.append([first, last])
        hanoi(n - 1, mid, last, first)

    answer = []
    hanoi(n, 1, 3, 2)
    print(answer)
    return answer
n = 3

solution(n)